transcendence-app/src/shared/authorization/casl-ability.factory.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  AbilityBuilder,
+  AnyMongoAbility,
+  createMongoAbility,
+} from '@casl/ability';
+import { Injectable } from '@nestjs/common';
+import { Action } from '../enums/action.enum';
+import { AuthUserCtxForChatroomMember } from './authorizationContext/authuser.chatroom-member';
+import { GlobalAuthUserCtx } from './authorizationContext/authuser.global';
+import { AuthUserCtxForChatroom } from './authorizationContext/authuser.chatroom';
+
+@Injectable()
+export class CaslAbilityFactory {
+  private readonly abilityCtx: AbilityBuilder<AnyMongoAbility>;
```

Comment:
probably it's not good to instantiate this at class build time. We need new ability instances on every call to defineAbilitiesFor... , and build() probably will return instances bound to this instance.